datasets/samplemotordatalimerick.py

```diff
@@ -294,205 +105,122 @@ def __download(self):
+        Reads csv files and returns file identifier, raw data,
+        sensor frequency, speed, load and label.
+        """
+        df_raw = pd.read_csv(file_full_path, sep=';', header=None)
+
+        df_raw.rename(
+            columns={0: 'Time', 1: 'Voltage_x', 2: 'Voltage_y',
+                     3: 'Voltage_z', 4: 'x', 5: 'y', 6: 'z'},
+            inplace=True
+            )
+        ss_vibr_x1 = df_raw.iloc[0]['x']
+        ss_vibr_y1 = df_raw.iloc[0]['y']
+        ss_vibr_z1 = df_raw.iloc[0]['z']
+        df_raw["Acceleration_x (g)"] = 50 * (df_raw["Voltage_x"] - ss_vibr_x1)
+        df_raw["Acceleration_y (g)"] = 50 * (df_raw["Voltage_y"] - ss_vibr_y1)
+        df_raw["Acceleration_z (g)"] = 50 * (df_raw["Voltage_z"] - ss_vibr_z1)
 
-            print('\nPickle files are already generated ...\n')
+        raw_data = df_raw[["Acceleration_x (g)", "Acceleration_y (g)", "Acceleration_z (g)"]]
+        raw_data = raw_data.to_numpy()
 
-            (self.signal_list, self.lbl_list) = \
-                pickle.load(open(self.dataset_pkl_file_path, 'rb'))
-            return
+        return [os.path.basename(file_full_path).split('/')[-1],
+                raw_data, sensor_sr_Hz, speed, load, label]
 
-        self.__gen_datasets()
+    def __getitem__(self, index):
+        if self.accel_in_second_dim:
+            signal, lbl = super().__getitem__(index)
+            signal = torch.transpose(signal, 0, 1)
+            lbl = lbl.transpose()
+            return signal, lbl
+        return super().__getitem__(index)
 
-    def normalize_signal(self, features):
+    def gen_dataframe(self):
         """
-        Normalize signal with Local Min Max Normalization
+        Generate dataframes from csv files of Sample Motor Data Limerick
         """
-        # Normalize data:
-        for instance in range(features.shape[0]):
-            instance_max = np.max(features[instance, :, :], axis=1)
-            instance_min = np.min(features[instance, :, :], axis=1)
-
-            for feature in range(features.shape[1]):
-                for signal in range(features.shape[2]):
-                    features[instance, feature, signal] = (
-                        (features[instance, feature, signal] - instance_min[feature]) /
-                        (instance_max[feature] - instance_min[feature])
-                    )
+        file_name = f'{self.__class__.__name__}_dataframe.pkl'
+        df_path = \
+            os.path.join(self.root, self.__class__.__name__, file_name)
+
+        if os.path.isfile(df_path):
+            print(f'\n{file_name} file already exists\n')
+            main_df = pd.read_pickle(df_path)
 
-        return features
+            return main_df
 
-    def __gen_datasets(self):
-        print('\nGenerating dataset pickle files from the raw data '
-              f'files (specs identifier: {self.specs_identifier}) ...\n')
+        print('\nGenerating dataset pickle files from the raw data \n')
```

Comment:
print message: not dataset pickle files but data frame indeed

Reply:
Updated